ruby/lib/ci/queue/redis/base.rb

```diff
@@ -58,6 +57,13 @@ def master_status
           redis.get(key('master-status'))
         end
 
+        def master_elected?
+          @master_elected ||= begin
+            status = master_status
+            status == 'ready' || status == 'finished'
```

Comment:
Do we consider the master elected after the queue is populated (state == ready) or when a worker gets the lock to begin pushing the queue (state == setup)? Just want to make sure `master_elected?` is what you intend.

Reply:
Technically, the master is indeed already elected when the status is set to `setup`.

However at that point the queue isn't filled, so yeah maybe that method should be along the lines of`master_elected_and_queue_initialized?` but it's seem a bit verbose.

WDYT?

Reply:
maybe just `queue_initialized?`.That implies that a master has been elected as well.

Reply:
Makes sense.